elia_chat/widgets/text_area.py

```diff
@@ -0,0 +1,45 @@
+"""
+Custom TextArea widget
+"""
+
+from dataclasses import dataclass
+
+from textual import events
+from textual.message import Message
+from textual.reactive import Reactive, reactive
+from textual.widgets import TextArea
+
+
+class EliaTextArea(TextArea):
+    """
+    Custom TextArea widget
+    """
+
```

Comment:
I believe that this can be extended to support the `shift+enter` functionality with something like:
```python
    BINDINGS.append(Binding("shift+enter", "new_line", "Begin a new line", show=False)
    # prefixing this function with `action_` appears to be critical
    def action_new_line(self) -> None:
        (select, center, record_width) = (False, True, False)
        cursor_row, cursor_column = self.cursor_location
        # I haven't tested `clamp_visitable`. I expect the document may need
        # to be extended, as the newline is probably at the end.
        target = self.clamp_visitable((current_row + 1, current_column + columns))
        self.move_cursor(target, select, center, record_width)
```

Reply:
😢 unfortunately `shift+enter` is a feature not supported at the terminal level https://github.com/Textualize/textual/discussions/3542. When the `enter` key is pressed, textual doesn't receive any other keys pressed with it.

Reply:
How about Ctrl+Enter